core/src/main/java/org/apache/iceberg/BaseReplacePartitions.java

```diff
@@ -79,6 +79,20 @@ public ReplacePartitions validateNoConflictingData() {
     return this;
   }
 
+  /**
+   * Validate the current metadata.
+   *
+   * <p>Child operations can override this to add custom validation.
+   *
+   * @param currentMetadata current table metadata to validate
+   * @deprecated Will be removed in 1.2.0, use {@link SnapshotProducer#validate(TableMetadata,
+   *     Snapshot)}.
+   */
+  @Deprecated
+  public void validate(TableMetadata currentMetadata) {
```

Comment:
we need to add separate Deprecations for `validate()` and `apply` here and in a few other places, because their visibility has been `public` and not `protected` (as it is defined in `SnapshotProducer`)